src/hotspot/share/gc/serial/serialHeap.cpp

```diff
@@ -672,8 +708,21 @@ void SerialHeap::do_full_collection(bool clear_all_soft_refs) {
   COMPILER2_OR_JVMCI_PRESENT(DerivedPointerTable::update_pointers());
 
   // Adjust generation sizes.
-  _old_gen->compute_new_size();
-  _young_gen->compute_new_size();
+  if (SharedSerialGCVirtualSpace) {
+    size_t young_gen_size_before = _shared_virtual_space->young_region().byte_size();
+    size_t tenured_gen_size = _old_gen->compute_new_size();
+    size_t young_gen_size = _young_gen->compute_new_size();
+
+    bool success = _shared_virtual_space->resize(tenured_gen_size, young_gen_size);
+    if (success) {
```

Comment:
What should happen if this fails?

Reply:
> What should happen if this fails?

I have added a comment that "execution will continue with the existing heap generation sizes if resizing the VirtualSpace fails"